driver/src/contracts/base_contract.rs

```diff
@@ -13,6 +21,10 @@ pub struct BaseContract {
     pub contract: Contract<web3::transports::Http>,
     pub web3: web3::Web3<web3::transports::Http>,
     event_loop: web3::transports::EventLoopHandle,
+    abi: ethabi::Contract,
+    network_id: u8,
+    public_key: H160,
+    private_key: H256,
```

Comment:
Will this private key be stored in the driver's memory? Is this something one would need to be cautious about when deploying a driver?

Reply:
I wouldn't be too concerned about memory. More importantly it will be an environment variable and thus also stored on disk somewhere. But I believe this is how most services work these days. Alternatively we could allow the private key to be entered on startup. But not sure that's worth the hassle.

Reply:
As we require storage of private keys and other sensitive information, are there any future plans to do a security audit of the crate dependencies we use?

Reply:
@anxolin what did we do in this regard for the dutchX. Did we have an audit for the services or the dependencies they require? I think our general strategy is to use accounts, which don't have a ton of eth and fund them as we go.

Reply:
Hi, I missed this comment since back then I didn't have the Gmail tagging rules I have now. Hopefully I'll miss less comments now. Let me know if I'm being unresponsive anytime, sometimes it can take me a bit to answer but I'll try not to miss this ones.

For DutchX we started simple, as you suggest, with accounts with limited amounts. We used a mnemonic, and we were using one account per bot. We use to have dozens of bots.

Then we added another layer of security by using the safe. We wrote a module that was allowing ONLY OUR ADDRES to trade ONLY SOME TOKENS and ONLY IN DUTCHX CONTRACT

So the module didn't allow to withdraw money, only trade